subprojects/core/src/main/java/org/gradle/initialization/DefaultCommandLineConverter.java

```diff
@@ -50,17 +52,12 @@
     private static final String PROJECT_CACHE_DIR = "project-cache-dir";
     private static final String RECOMPILE_SCRIPTS = "recompile-scripts";
 
-    private static final String CONFIGURE_ON_DEMAND = "configure-on-demand";
-
     private static final String CONTINUOUS = "continuous";
     private static final String CONTINUOUS_SHORT_FLAG = "t";
 
     private static final String BUILDSCAN = "scan";
```

Comment:
It seems like we ought to be able to apply this to `--scan` as well. We want consistency, so we should also make `--scan` and `--no-scan` behave the same as all other boolean options do.

Reply:
This lines up with my comment above:

> There's also the case of a command line option that does not have a Gradle property representation. Should we define those option with the same data structure? At the moment the assumption is that a Gradle property is mandatory and the command line option is optional.

I guess either the Gradle property _or_ the command line option can be optional. That would slightly change the API.

Reply:
Scan should be a build option that behaves consistently with the other build options.